frame/contracts/src/lib.rs

```diff
@@ -1172,6 +1189,11 @@ impl<T: Config> Pallet<T> {
 			debug_message: debug_message.as_mut(),
 		};
 		let output = InstantiateInput::<T> { code, salt }.run_guarded(common);
+		let events = if collect_events {
+			Some(System::<T>::read_events_no_consensus().map(|e| *e).collect())
```

Comment:
nit (check it compiles but you can use the more idiomatic copied here I believe
```suggestion
			Some(System::<T>::read_events_no_consensus().copied().collect())
```

Reply:
it would be nice but it does not compile :(